lib/internal/modules/esm/translators.js

```diff
@@ -29,9 +32,19 @@ module.exports = translators;
 translators.set('esm', async (url) => {
   const source = `${await readFileAsync(new URL(url))}`;
   debug(`Translating StandardModule ${url}`);
+  const module = new ModuleWrap(stripShebang(source), url);
+  callbackMap.set(module, {
+    initializeImportMeta(meta) {
+      meta.url = url;
+    },
+    async importModuleDynamically(specifier) {
+      const loader = await esmLoader.loaderPromise;
+      return loader.import(specifier, url);
+    },
+  });
```

Comment:
Do these functions need to be closures? Could we not have them be globally/loader-level defined with another argument - 

```js
initializeImportMeta(meta, url) {
  meta.url = url
},
importModuleDynamically: loader.import.bind(loader)
```

Reply:
that seems reasonable